bin/format-code.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+TMP_SOURCES_FILE=$(mktemp)
+
+find ./* -type f \( -name '*.h' -o -name '*.cpp' \) \
+     -not \( -path './testing/googletest/*' -o -name 'generated-testcase.cpp' \) \
+     -print0 > "${TMP_SOURCES_FILE}"
+
+xargs -0 clang-format -i \
+      < "${TMP_SOURCES_FILE}"
+
+# shellcheck disable=SC2016
+xargs -0 perl -pi -e 's/^(__attribute__\(\(\w*\)\)) (\w)/$1\n$2/' \
```

Comment:
While I agree that the change this transform makes is desirable, I think it's outweighed by the value of having clang-format "just work" , even if it's run by, say, an IDE's tools

Reply:
I find the `__attribute__((weak))` at the start of the line really hides the declaration from my eyes, so I have a strong preference for putting it on the line above, but I take your point.  Having just played with setting up VS Code to use clang-format on Kaleidoscope last night, I do have two alternative suggestions for your consideration, but I'm okay with whatever you decide.  Here are the options I've thought of:

1. Just let clang-format do its thing, and maybe someday it'll have an option for line breaks after leading attribute macros.
2. Add a single-file version of this script that first calls clang-format, then makes further adjustments that we like, and configure IDEs to call that script instead of clang-format directly.  This is what I would do for my own project.
3. Switch from leading attribute declarations to trailing ones.  I think this works, but the docs I found from gcc are unclear.  I would prefer this to option 1:
```c++
void strongFunc();
bool weakFunction(int test) __attribute__((weak));
// vs
void strongFunc();
__attribute__((weak)) bool weakFunction(int test);
```

Reply:
GitHub's syntax highlighter made the second one look worse than it really is, but I still have a strong preference for trailing attributes if they have to be on the same line (assuming that actually works properly).

Reply:
I have one more possible option:
```
void strongFunc();
__attribute__((weak))  //
bool weakFunction(int test);
```

If this doesn't float your boat, I think I lean toward trailing attributes.

Reply:
I know I tried that at one point, but I thought there was some problem with it.  I may be confusing it with some other stuff I tried, though.

Reply:
Yeah, for some mysterious reason, the comment trick doesn't work in this case.  We get this instead:
```c++
__attribute__((weak))  //
bool
weakFunction(int test);
```

Reply:
Regardless, we can kick that style issue down the road.

Reply:
Happy to kick this one down the road.